### PR TITLE
Disable "Next" button in Paginator when the next page is empty

### PR DESCRIPTION
**Pull request by noskla**
## Description
It may cause confusion to the user why pages in the community list are empty when there are no more communities to display. This small change fixes this issue by disabling the Next button if the number of fetched communities is less than the limit.

Additionally, I have moved the communityLimit constant to the config file.

## Possible issues with this approach
The Next button will be still enabled, if there are exactly _{communityLimit}_ communities on the instance, to mitigate this the API would have to tell the client if currently viewed page is last, or the client could fetch next page in advance.

### CODE REVIEW

**Review by SleeplessOne1917 — APPROVED** [review]
This is a good fix. It's probably worth doing something like this everywhere the paginator is used.

File: src/shared/components/community/communities.tsx [hunk 1]
```diff
@@ -221,7 +221,14 @@ export class Communities extends Component<any, CommunitiesState> {
                 </tbody>
               </table>
             </div>
-            <Paginator page={page} onChange={this.handlePageChange} />
+            <Paginator
+              page={page}
+              onChange={this.handlePageChange}
+              nextDisabled={
+                communityLimit >
+                this.state.listCommunitiesResponse.data.communities.length
```
Comment (dessalines) [inline src/shared/components/community/communities.tsx:229]: Looks good. Could you add this logic also everywhere else that its feasible? (IE Home.tsx, Community.tsx especially) 

The pages with multiple request types (like `inbox.tsx`, `modlog.tsx` , it won't work, so you can ignore those ones)

**Review by dessalines — APPROVED** [review]
Looks good, thx!

File: src/shared/components/common/paginator.tsx [hunk 2]
```diff
@@ -23,6 +24,7 @@ export class Paginator extends Component<PaginatorProps, any> {
         <button
           className="btn btn-secondary"
           onClick={linkEvent(this, this.handleNext)}
+          disabled={this.props.nextDisabled || false}
```
Comment (SleeplessOne1917) [inline src/shared/components/common/paginator.tsx:27]: Is the `|| false` necessary here?

File: src/shared/components/home/emojis-form.tsx [hunk 3]
```diff
@@ -267,7 +267,11 @@ export class EmojiForm extends Component<EmojiFormProps, EmojiFormState> {
             {I18NextService.i18n.t("add_custom_emoji")}
           </button>
 
-          <Paginator page={this.state.page} onChange={this.handlePageChange} />
+          <Paginator
+            page={this.state.page}
+            onChange={this.handlePageChange}
+            nextDisabled={false}
```
Comment (SleeplessOne1917) [inline src/shared/components/home/emojis-form.tsx:273]: Is there never a time that this needs to be disabled?

File: src/shared/components/modlog.tsx [hunk 4]
```diff
@@ -860,7 +860,11 @@ export class Modlog extends Component<
               </thead>
               {this.combined}
             </table>
-            <Paginator page={page} onChange={this.handlePageChange} />
+            <Paginator
+              page={page}
+              onChange={this.handlePageChange}
+              nextDisabled={false}
```
Comment (SleeplessOne1917) [inline src/shared/components/modlog.tsx:866]: Same here.

File: src/shared/components/common/paginator.tsx (hunk 2 above)
Comment (noskla) [reply src/shared/components/common/paginator.tsx:27]: Good catch, it's a residue after nextDisabled was optional to not pass "undefined" to the property, I'll fix it

File: src/shared/components/home/emojis-form.tsx (hunk 3 above)
Comment (dessalines) [reply src/shared/components/home/emojis-form.tsx:273]: It needs to be disabled when there are multiple lists (because the API doesn't combine them)

**Review by SleeplessOne1917 — APPROVED** [review]

**Pull request merged**